r/geoarrow/R/vctr.R

```diff
@@ -28,16 +41,80 @@ new_geoarrow_vctr <- function(chunks, schema, indices = NULL) {
 `[.geoarrow_vctr` <- function(x, i) {
   attrs <- attributes(x)
   x <- NextMethod()
-  # Assert slice?
+
+  if (is.null(vctr_as_slice(x))) {
+    stop(
+      "Can't subset geoarrow_vctr with non-slice (e.g., only i:j indexing is supported)"
+    )
+  }
+
   attributes(x) <- attrs
   x
 }
 
+#' @export
+`[<-.geoarrow_vctr` <- function(x, i, value) {
+  stop("subset assignment for geoarrow_vctr is not supported")
+}
+
+#' @export
+`[[<-.geoarrow_vctr` <- function(x, i, value) {
+  stop("subset assignment for geoarrow_vctr is not supported")
+}
+
+#' @export
+format.geoarrow_vctr <- function(x, ..., width = NULL, digits = NULL) {
+  if (is.null(width)) {
+    width <- getOption("width", 100L)
+  }
+
+  width <- max(width, 20)
+
+  if (is.null(digits)) {
+    digits <- getOption("digits", 7L)
+  }
+
+  digits <- max(digits, 0)
+
+  formatted_array <- geoarrow_kernel_call_scalar(
+    "format_wkt",
+    x,
+    options = c(
+      max_element_size_bytes = width - 10L,
+      precision = digits
+    ),
+    n = length(attr(x, "chunks"))
```

Comment:
Prefer exact = TRUE? Applies to other attribute getters across the package.

```suggestion
    n = length(attr(x, "chunks", exact = TRUE))
```